### PR TITLE
Handle header trees starting with a non-H1 header 

### PR DESCRIPTION
**Pull request by russellkan**
# Describe Your Changes
- Avoid `IndexError` when header trees start with a non-H1 header
- Skip out of order headers with a warning
- Add unit test cases

# How Did You Test It
-  Ran `pytest tests` with all tests passing

### CODE REVIEW

**Review by bimbashrestha — APPROVED** [review]
Looks good to me! @sHermanGriffiths, please review/approve this when you have a chance. 

@russellkan feel free to merge after Sam'ans approval.

**Review by sHermanGriffiths — APPROVED** [review]
Good stuff Russel! Looks great to me!

**Pull request merged**
